sdcm/nemesis.py

```diff
@@ -434,6 +437,135 @@ def disrupt_stop_start_scylla_server(self):  # pylint: disable=invalid-name
+        #
+        # def _start_up_nodes():
+        #     for node in self.cluster.nodes:
+        #         node.start_scylla()
+        #         node.wait_db_up()
+        #         node.wait_jmx_up()
+
+        # _flush_and_stop_nodes()
+        # _downgrade_sstables_format()
+        # _start_up_nodes()
+
+        compaction_ops = CompactionOps(self.cluster)
+        trigger_func = {"func": compaction_ops.trigger_upgrade_compaction, "kwargs": {}}
+        watch_func = {"func": compaction_ops.stop_on_user_compaction_logged,
+                      "kwargs": {"node": node1,
+                                 "watch_for": "Upgrade keyspace1.standard1",
+                                 "timeout": 300,
+                                 "stop_func": compaction_ops.stop_upgrade_compaction}}
+        self._trigger_and_stop_compaction(trigger_func_w_kwargs=trigger_func, watch_func_w_kwargs=watch_func)
+
+    def disrupt_start_stop_reshape_compaction(self):
+        node = self.cluster.nodes[0]
+        compaction_ops = CompactionOps(self.cluster)
+
+        def _trigger_reshape(node: BaseNode, tester: ClusterTester, keyspace: str = "keyspace1"):
+            twcs = {'class': 'TimeWindowCompactionStrategy', 'compaction_window_size': 1,
+                    'compaction_window_unit': 'MINUTES', 'max_threshold': 1, 'min_threshold': 1}
+            node.run_nodetool(sub_cmd="flush")
+            tester.wait_no_compactions_running()
+            LOGGER.info("Copying data files to ./staging and ./upload directories...")
+            keyspace_dir = f'/var/lib/scylla/data/{keyspace}'
+            cf_data_dir = node.remoter.run(f"ls {keyspace_dir}").stdout.splitlines()[0]
+            full_dir_path = f"{keyspace_dir}/{cf_data_dir}"
+            upload_dir = f"{full_dir_path}/upload"
+            staging_dir = f"{full_dir_path}/staging"
+            cp_cmd_upload = f"cp -p {full_dir_path}/md-* {upload_dir}"
+            cp_cmd_staging = f"cp -p {full_dir_path}/md-* {staging_dir}"
+            node.remoter.sudo(cp_cmd_staging)
+            node.remoter.sudo(cp_cmd_upload)
+            LOGGER.info("Finished copying data files to ./staging and ./upload directories.")
+            cmd = f"ALTER TABLE standard1 WITH compaction={twcs}"
+            node.run_cqlsh(cmd=cmd, keyspace="keyspace1")
+            node.run_nodetool("refresh -- keyspace1 standard1")
+        trigger_func = {"func": _trigger_reshape, "kwargs": {"node": node, "tester": self.tester}}
+        watch_func = {"func": compaction_ops.stop_on_user_compaction_logged,
+                      "kwargs": {"node": node,
+                                 "watch_for": "Upgrade keyspace1.standard1",
+                                 "timeout": 300,
+                                 "stop_func": compaction_ops.stop_upgrade_compaction}}
+        self._trigger_and_stop_compaction(trigger_func_w_kwargs=trigger_func, watch_func_w_kwargs=watch_func)
+
+    def _trigger_and_stop_compaction(self,
```

Comment:
move this function into Parallel object, call it `run_once`, make it class method
Use functools.partial to pass function and it's parameters to it

Reply:
Good idea. I created a `run_once` method and switched the nemesis methods to pass partial objects instead of dicts to it.